### PR TITLE
Darknet_CUDA - Darknet with CUDA support

### PR DESCRIPTION
**Pull request by ianshmean**
Continuation of #1122 but with CUDA support

### CODE REVIEW

File: D/Darknet_CUDA/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+script = raw"""
+cd $WORKSPACE/srcdir
+
+## CUDA setup
+
+mv cudnn $prefix                # Note - when MacOS is added, Darknet expects cudnn in cuda's dir: https://github.com/AlexeyAB/darknet/blob/master/Makefile#L112
+mkdir -p /usr/local/cudnn
+cd /usr/local/cudnn
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:22]:
```suggestion
```

File: D/Darknet_CUDA/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ln -s ${prefix}/cudnn /usr/local/cudnn
+
+mv cuda $prefix
+mkdir -p /usr/local/cuda
+cd /usr/local/cuda
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:27]:
```suggestion
```

File: D/Darknet_CUDA/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+
+
+script = raw"""
+cd $WORKSPACE/srcdir
+
+## CUDA setup
+
+mv cudnn $prefix                # Note - when MacOS is added, Darknet expects cudnn in cuda's dir: https://github.com/AlexeyAB/darknet/blob/master/Makefile#L112
+mkdir -p /usr/local/cudnn
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:20]:
```suggestion
mkdir -p /usr/local
```

File: D/Darknet_CUDA/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+script = raw"""
+cd $WORKSPACE/srcdir
+
+## CUDA setup
+
+mv cudnn $prefix                # Note - when MacOS is added, Darknet expects cudnn in cuda's dir: https://github.com/AlexeyAB/darknet/blob/master/Makefile#L112
+mkdir -p /usr/local
+ln -s ${prefix}/cudnn /usr/local/cudnn
+ln -s ${prefix}/cuda /usr/local/cuda
+export PATH=$PATH:/usr/local/cuda
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:23]:
```suggestion
export PATH=$PATH:/usr/local/cuda/bin
```

File: D/Darknet_CUDA/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+script = raw"""
+cd $WORKSPACE/srcdir
+
+## CUDA setup
+
+mv cudnn $prefix                # Note - when MacOS is added, Darknet expects cudnn in cuda's dir: https://github.com/AlexeyAB/darknet/blob/master/Makefile#L112
+mkdir -p /usr/local
+ln -s ${prefix}/cudnn /usr/local/cudnn
+ln -s ${prefix}/cuda /usr/local/cuda
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:22]:
```suggestion
ln -s ${prefix}/cuda /usr/local/cuda
# Darknet expects the lib in `/usr/local/cuda/lib`: https://github.com/AlexeyAB/darknet/blob/cde72f8e0bc177439c32ce49740fd45ebc79859a/Makefile#L113
ln -s lib64 /usr/local/cuda/lib
```

File: D/Darknet_CUDA/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+]
+
+
+
+script = raw"""
+cd $WORKSPACE/srcdir
+
+## CUDA setup
+
+mv cudnn $prefix                # Note - when MacOS is added, Darknet expects cudnn in cuda's dir: https://github.com/AlexeyAB/darknet/blob/master/Makefile#L112
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:19]:
```suggestion
mv cudnn $prefix                # Note - when MacOS is added, Darknet expects cudnn in cuda's dir: https://github.com/AlexeyAB/darknet/blob/cde72f8e0bc177439c32ce49740fd45ebc79859a/Makefile#L112
```

File: D/Darknet_CUDA/build_tarballs.jl [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+]
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libdarknet", :libdarknet; dont_dlopen = true),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency(PackageSpec(name="CUDA_full_jll", version=v"10.1.243")),
```
Comment (giordano) [inline D/Darknet_CUDA/build_tarballs.jl:77]: Why we're restricting to this specific version of CUDA?

Comment (ianshmean) [reply D/Darknet_CUDA/build_tarballs.jl:77]: I just copied what torch had. I don’t see why we need to

Comment (giordano) [reply D/Darknet_CUDA/build_tarballs.jl:77]: @maleadt do you think it's important to specify a cuda version? Are there important incompatibilities across versions?

Comment (maleadt) [reply D/Darknet_CUDA/build_tarballs.jl:77]: Depends on the software. Libraries like Torch are very specific on CUDA support, but an application like Darknet may be more lenient. Although the README specifies CUDA=10.0

**Pull request merged**